model/Bom.js

```diff
@@ -38,11 +41,47 @@ class Bom extends CycloneDXObject {
     if (pkg) {
       this._metadata = this.createMetadata(pkg, componentType)
       this._components = this.listComponents(pkg, lockfile)
+      this._dependencies = this.listDependencies(pkg)
     } else {
       this._components = []
+      this._dependencies = []
     }
   }
 
+  listDependencies (pkg) {
+    const list = []
+    this.createDependency(pkg, list)
+    return list
+  }
+
+  createDependency (pkg, list) {
+    // read-installed with default options marks devDependencies as extraneous
+    // if a package is marked as extraneous, do not include it as a dependency
+    // if -d or --include-dev is specified then all dev dependencies are considered
+    if (pkg.extraneous) return
+    const ref = this.createPackageURL(pkg)
+    const deplist = []
+    if (pkg._dependencies && Object.keys(pkg._dependencies).length !== 0) {
+      Object.keys(pkg._dependencies)
+        .map(x => pkg.dependencies[x])
+        .filter(x => x !== undefined && typeof (x) !== 'string') // remove cycles
+        .map(x => deplist.push(new Dependency(this.createPackageURL(x), this.createDependency(x, list))))
```

Comment:
:question: 
how do you handle, if the return of `this.createPackageURL()` return `null`? which is totally possible from the current algorithm used?

Reply:
Its not handled yet and as I said in the previous comment, creation of BomReference class will solve this issue.

Reply:
see https://github.com/CycloneDX/cyclonedx-node-module/pull/244#issuecomment-1024994973